Source/CarthageKit/VersionFile.swift

```diff
@@ -0,0 +1,221 @@
+		
+		var frameworks = frameworksByPlatform[platformName] ?? []
+		frameworks.append(cachedFramework)
+		frameworksByPlatform[platformName] = frameworks
+	}
+	
+	var cachedPlatforms: [String: CachedPlatform] = frameworksByPlatform.reduce([:]) { (cachedPlatforms, entry: (platformName: String, cachedFrameworks: [CachedFramework])) in
+		var cachedPlatforms = cachedPlatforms
+		cachedPlatforms[entry.platformName] = CachedPlatform(commitish: dependency.version.commitish, cachedFrameworks: entry.cachedFrameworks)
+		return cachedPlatforms
+	}
+	
+	let rootBinariesURL = rootDirectoryURL.appendingPathComponent(CarthageBinariesFolderPath, isDirectory: true).URLByResolvingSymlinksInPath!
+	let versionFileURL = rootBinariesURL.appendingPathComponent(".\(dependency.project.name).version")
+	
+	if let oldVersionFile: VersionFile = readVersionFileAtURL(versionFileURL) {
+		for platform in Platform.supportedPlatforms {
+			if let oldCachedPlatform = oldVersionFile.cacheForPlatform(platform) where cachedPlatforms[platform.rawValue] == nil {
+				cachedPlatforms[platform.rawValue] = oldCachedPlatform
+			}
+		}
+	}
+	
+	let versionFile = VersionFile(
+		macOS: cachedPlatforms[Platform.Mac.rawValue],
+		iOS: cachedPlatforms[Platform.iOS.rawValue],
+		watchOS: cachedPlatforms[Platform.watchOS.rawValue],
+		tvOS: cachedPlatforms[Platform.tvOS.rawValue])
+	
+	do {
+		let json = versionFile.toJSONObject()
+		let jsonData = try NSJSONSerialization.dataWithJSONObject(json, options: .PrettyPrinted)
+		try jsonData.writeToURL(versionFileURL, options: .DataWritingAtomic)
+	}
+	catch {
+		return false
+	}
+	
+	return true
+}
+
+/// Determines whether a dependency can be skipped.  If a version file
+/// for the dependency project exists, and its commitish matches, and
+/// the recorded SHA1s are the same as the computed SHA1s of each
+/// framework in the Carthage/Build directory for the given platforms,
+/// the dependency can be skipped.
+///
+/// If a set of platforms is not provided and a version file exists,
+/// the platforms listed in the version file are used instead.
+///
+/// Returns true if the the dependency can be skipped.
+public func versionFileMatchesDependency(dependency: Dependency<PinnedVersion>, forPlatforms platforms: Set<Platform>, rootDirectoryURL: NSURL) -> Bool {
+	let rootBinariesURL = rootDirectoryURL.appendingPathComponent(CarthageBinariesFolderPath, isDirectory: true).URLByResolvingSymlinksInPath!
+	let versionFileURL = rootBinariesURL.appendingPathComponent(".\(dependency.project.name).version")
+	guard let versionFile: VersionFile = readVersionFileAtURL(versionFileURL) else { return false }
+	let commitish = dependency.version.commitish
+	
+	let cachedPlatforms = versionFile.cachedPlatforms()
+	let platformsToCheck = platforms.isEmpty ? cachedPlatforms : platforms
+	for platform in platformsToCheck {
+		if !checkPlatform(platform, versionFile: versionFile, commitish: commitish, rootDirectoryURL: rootDirectoryURL) {
+			return false
+		}
+	}
```

Comment:
The work of checking whether a platform can be skipped should be in a method on `VersionFile`.